docs/terraform/common-troubleshooting.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: "Common Troubleshooting & Recipes"
+description: "Learn about overcoming API design incompatibilities when generating Terraform providers with Speakeasy."
+---
+
+# Common Troubleshooting & Recipes
+
+## Overview
+
+When generating Terraform providers from OpenAPI specifications, you might
+encounter cases where your API’s structure doesn’t naturally fit Terraform’s
+resource-oriented approach. These differences typically occur because APIs
+aren’t always designed with Terraform’s infrastructure management style in
+mind.
+
+Speakeasy’s generator identifies these design differences and offers extensions
+and customization options to address them. This enables you to produce
+effective Terraform providers, even when your API doesn’t initially match
+Terraform’s requirements, simplifying what might otherwise be challenging
+configuration tasks.
+
+## Fixing Common API Issues
+
+### Impedance Mismatch Errors
+
+An impedance mismatch error happens when Speakeasy’s generator finds properties
+with different data types that need to be combined into one. This error means
+the request and response data don’t match up correctly, which can cause
```

Comment:
Nit: I think we should mention that this data mismatch can be _across_ API operations, not just request and response of a single API operation 👍